cmd/setup.go

```diff
@@ -304,3 +308,39 @@ func configureLoadPoints(conf config, cp *ConfigProvider) (loadPoints []*core.Lo
 
 	return loadPoints, nil
 }
+
+func configureCircuits(site *core.Site, loadPoints []*core.LoadPoint, cp *ConfigProvider) (err error) {
+	ccInterfaces, _ := viper.AllSettings()["circuits"].([]interface{})
+
+	for _, ccI := range ccInterfaces {
+		var ccMap map[string]interface{}
+		if err := util.DecodeOther(ccI, &ccMap); err != nil {
+			return fmt.Errorf("failed decoding circuit configuration: %w", err)
+		}
+
+		ccNew, err := core.NewCircuitFromConfig(cp, ccMap, site)
+		if err != nil {
+			return fmt.Errorf("failed configuring circuit: %w", err)
+		}
+		site.Circuits = append(site.Circuits, ccNew)
+	}
+	// connect circuits and lps
+	for lpId := range loadPoints {
```

Comment:
for _, lp :=

Reply:
wenn ich das mit dem lp mache, dann verliere ich die Änderungen am LP welche innerhalb der Loop gemacht werden (copy everywhere). Mit [lpId] werden Referenzen benutzt.

Reply:
Das passiert nur dann wenn der Loadpoint kein Pointer oder Interface ist. Also lieber das ändern wenn notwendig.

Reply:
Sind pointer, aber so war es der einzige Weg das hinzubekommen `var loadPoints []*core.LoadPoint`

Reply:
> so war es der einzige Weg das hinzubekommen

Was meinst Du damit- um was hinzubekommen? Das Pattern den Index zu nutzen braucht es nur wenn das zweite Range Ergebnis nicht per Referenz übergeben wird. Bei Pointern und Interfaces ist das aber immer der Fall.